uid2_client/publisher_client.py

```diff
@@ -50,12 +50,14 @@ def generate_token(self, token_generate_input):
         req, nonce = make_v2_request(self._secret_key, dt.datetime.now(tz=timezone.utc),
                                      token_generate_input.get_as_json_string().encode())
         resp = post(self._base_url, '/v2/token/generate', headers=auth_headers(self._auth_key), data=req)
-        resp_body = parse_v2_response(self._secret_key, resp.read(), nonce)
+        resp.raise_for_status()
```

Comment:
In future we should probably wrap `base_url`, `auth_key`, `secret_key` in their own class to get better reuse / single responsibility. 
That class could have a method that replaces lines 50-54 with a single line (as those lines appear often):

`resp_body = v2_request_helper.request('/v2/token/generate', token_generate_input.get_as_json_string())`